File: pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    L = count_letters(text)
    W = count_words(text)
    S = count_sentences(text)

    index = 0.0588 * L / W * 100 - 0.296 * S / W * 100 - 15.8
    index = index + 1 if index - int(index) >= 0.5 else index

    if index >= 16:
        print("Grade 16+")
    elif index >= 1:
        print(f"Grade {int(index)}")
    else:
        print("Before Grade 1")


def count_letters(text):
    count = 0
    for char in text:
        if char.isalpha():
            count += 1

    return count


def count_words(text):
    count = 1
    for i in range(1, len(text)):
        if text[i].isalpha() and not text[i - 1].isalpha() and text[i - 1] != "-" and text[i - 1] != "'":
            count += 1

    return count


def count_sentences(text):
    count = 0
    for char in text:
        if char == "." or char == "!" or char == "?":
            count += 1

    return count


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");
    int L = count_letters(text);
    int W = count_words(text);
    int S = count_sentences(text);

    float index = 0.0588 * L / W * 100 - 0.296 * S / W * 100 - 15.8;
    index = (index - (int) index >= 0.5) ? (int) index + 1 : index;

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index >= 1)
    {
        printf("Grade %i\n", (int) index);
    }
    else
    {
        printf("Before Grade 1\n");
    }

}

int count_letters(string text)
{
    int count = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    //printf("%i letter(s)\n", count);
    return count;
}

int count_words(string text)
{
    int count = 1;

    for (int i = 1, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]) && !isalpha(text[i - 1]) && text[i - 1] != '-' && text[i - 1] != '\'')
        {
            count++;
        }
    }
    //printf("%i word(s)\n", count);
    return count;
}

int count_sentences(string text)
{
    int count = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
    //printf("%i sentence(s)\n", count);
    return count;
}